courseraresearchexports/models/ExportDb.py

```diff
@@ -0,0 +1,130 @@
+#!/usr/bin/env python
+
+# Copyright 2016 Coursera
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import csv
+
+from sqlalchemy import create_engine
+from sqlalchemy.engine import reflection
+
+from courseraresearchexports.models.ContainerInfo import ContainerInfo
+
+
+class ExportDb:
+    """
+    Interface to SQLAlchemy for interacting with the a database
+    on a docker container
+    """
+    def __init__(self, host_ip=None, host_port=None, db=None, **kwargs):
+
+        if not (host_ip and host_port and db):
+            raise ValueError(
+                'Host IP, port and database name must be specified')
+
+        self.host_ip = host_ip
+        self.host_port = host_port
+        self.db = db
+        self.engine = create_engine(
+            "postgresql://{user}@{host}:{port}/{db}"
+            .format(user='postgres',
+                    host=self.host_ip,
+                    port=self.host_port,
+                    db=self.db))
+
+    @classmethod
+    def from_container(cls, container_name_or_id, docker_client):
+        """
+        Create ExportDb object directly from container_name_or_id identifier.
+        :param container_name_or_id:
+        :param docker_client:
+        :return:
+        """
+        container_info = ContainerInfo.from_container(container_name_or_id,
+                                                      docker_client)
+        return cls.from_container_info(container_info)
+
+    @classmethod
+    def from_container_info(cls, container_info):
```

Comment:
You're right. Thought I had some usage of this method elsewhere, but looks like I completely refactored it out.